common/src/main/java/io/netty/util/DomainWildcardMappingBuilder.java

```diff
@@ -83,6 +83,9 @@ private String normalizeHostName(String hostname) {
         }
         hostname = ImmutableDomainWildcardMapping.normalize(checkNotNull(hostname, "hostname"));
         if (hostname.charAt(0) == '*') {
+            if (hostname.charAt(1) != '.' || hostname.length() < 3) {
```

Comment:
The order is reversed.   This might throw an NPE

Reply:
I am an idiot... fixed